ext/search_kit/search_kit.php

```diff
@@ -109,3 +109,39 @@ function search_kit_civicrm_post($op, $entity, $id, $object) {
     \Civi::cache('metadata')->clear();
   }
 }
+
+/**
+ * Implements hook_civicrm_entityTypes().
+ */
+function search_kit_civicrm_entityTypes(array &$entityTypes): void {
+  foreach (_search_kit_get_entity_displays() as $display) {
```

Comment:
@totten the "I-have-no-idea-how-they-are-related" test failures in this PR are driving me nuts, I'd like to pick your brain about it. Here's what I've found out.

1. The failing tests are all APIv4 tests that query `RelationshipCache` and fail to find their expected results.
2. So presumably the cause of the failures is missing triggers to populate the `civicrm_relationship_cache` table.
3. Putting an early return in this function fixes it and tests pass.
4. Drilling down into the `_search_kit_get_entity_displays` function, commenting out the query to fetch search displays also solves the issue (note that my 2nd commit tried changing it from `DAO::executeQuery()` to `$dao->fetch()`, to no avail).
5. Note that there are no search displays of type 'entity' so in the test environment that query never returns anything, and thus this `hook_entityTypes()` also doesn't do anything.
6. So it appears the very act of performing a `SELECT FROM civicrm_search_display` query during the bootstrap process causes the `RelationshipCache` trigger to be lost???

This makes no sense at all to me. Do you have any thoughts?

Reply:
Figured it out. Adding try/catch around the query fixes the problem. It's probably an early bootstrap issue in the test when the extension gets enabled but the table hasn't been added yet.